cmd/timescale-prometheus/main.go

```diff
@@ -3,265 +3,34 @@
 // LICENSE for a copy of the license.
 package main
 
-// Based on the Prometheus remote storage example:
-// documentation/examples/remote_storage/remote_storage_adapter/main.go
-
 import (
-	"context"
 	"flag"
 	"fmt"
-	"net/http"
-	pprof "net/http/pprof"
 	"os"
-	"regexp"
-	"strings"
-	"sync/atomic"
-	"time"
 
-	"github.com/jackc/pgx/v4/pgxpool"
-	_ "github.com/jackc/pgx/v4/stdlib"
-	"github.com/jamiealquiza/envy"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
-	"github.com/timescale/timescale-prometheus/pkg/api"
 	"github.com/timescale/timescale-prometheus/pkg/log"
-	"github.com/timescale/timescale-prometheus/pkg/pgclient"
-	"github.com/timescale/timescale-prometheus/pkg/pgmodel"
-	"github.com/timescale/timescale-prometheus/pkg/util"
+	"github.com/timescale/timescale-prometheus/pkg/timescale_prometheus"
 	"github.com/timescale/timescale-prometheus/pkg/version"
 )
 
-type config struct {
-	listenAddr        string
-	telemetryPath     string
-	pgmodelCfg        pgclient.Config
-	logLevel          string
-	haGroupLockID     int
-	restElection      bool
-	prometheusTimeout time.Duration
-	electionInterval  time.Duration
-	migrate           bool
-	stopAfterMigrate  bool
-	corsOrigin        *regexp.Regexp
-}
-
-const (
-	promLivenessCheck = time.Second
-)
-
-var (
-	elector *util.Elector
-)
-
 func main() {
-	cfg, err := parseFlags()
+	logLevel := flag.String("log-level", "debug", "The log level to use [ \"error\", \"warn\", \"info\", \"debug\" ].")
+	cfg := &timescale_prometheus.Config{}
+	cfg, err := timescale_prometheus.ParseFlags(cfg)
 	if err != nil {
 		fmt.Println("Version: ", version.Version, "Commit Hash: ", version.CommitHash)
 		fmt.Println("Fatal error: cannot parse flags ", err)
 		os.Exit(1)
 	}
-	err = log.Init(cfg.logLevel)
+	err = log.Init(*logLevel)
 	if err != nil {
 		fmt.Println("Version: ", version.Version, "Commit Hash: ", version.CommitHash)
 		fmt.Println("Fatal error: cannot start logger", err)
 		os.Exit(1)
 	}
-	log.Info("msg", "Version:"+version.Version+"; Commit Hash: "+version.CommitHash)
-	log.Info("config", util.MaskPassword(fmt.Sprintf("%+v", cfg)))
-
-	promMetrics := api.InitMetrics()
-	elector, err = initElector(cfg, promMetrics)
-
+	err = timescale_prometheus.Run(cfg)
```

Comment:
Are we planning to run multiple instances of timescale-prometheus? If not, then this seems wierd since timescale-prometheus's main file does not have the main code rather, some file in pkg has.

Reply:
If yes, then lets make this line and the following line in a single line.

Reply:
No. I still think inlining non-trivial functions into the error handling conditional is bad style